app/Main.hs

```diff
@@ -82,7 +86,7 @@ storageParser = initialStorage
   <*> fa2ConfigParser
   <*> contractMetadataConfigParser
   where
-    adminParser = (#admin :!) <$> addressOption Nothing
+    adminParser = ((#admin :!) . MkAddress) <$> (addressOption @'AddressKindImplicit) Nothing
```

Comment:
This is a bit of a contentious point. So _technically_ the contract itself doesn't say anything about whether `admin` is an implicit address, it could just as well work if `admin` is a smart contract. We had a bit of a discussion about this wrt tzbtc. Now, whether having a smart contract as admin is a realistic scenario is an open question, but it's not technically impossible, so it arguably makes sense to support this use case.

The catch is, we overlooked it in morley, so there isn't a convenient CLI parser for this. It's more or less straightforward to add one for the time being, see e.g. [tzbtc implementation](https://github.com/tz-wrapped/tezos-btc/blob/b9ec7d83139137e152f7883b7fc1087f4312af7b/src/CLI/L1AddressOrAlias.hs#L114)

Reply:
I Have changed `admin` arguments to `L1Address` and used the instance from Tzbtc implementation to parse the address as `L1Address`.